### PR TITLE
Update lint rules to match @peggyjs/eslint-config@3.1.0. 

### PR DESCRIPTION
**Pull request by hildjj**
Still need to update package.json and package-lock.json once that version is shipped.

### CODE REVIEW

File: bin/peggy-cli.js [hunk 1]
```diff
@@ -203,9 +213,9 @@ class PeggyCLI extends Command {
         new Option("-O, --optimize <style>")
           .hideHelp()
           .argParser(() => {
-            this.print(this.std.err, "Option --optimize is deprecated from 1.2.0 and has no effect anymore.");
-            this.print(this.std.err, "It will be deleted in 2.0.");
-            this.print(this.std.err, "Parser will be generated in the former \"speed\" mode.");
+            print(this.std.err, "Option --optimize is deprecated from 1.2.0 and has no effect anymore.");
+            print(this.std.err, "It will be deleted in 2.0.");
+            print(this.std.err, "Parser will be generated in the former \"speed\" mode.");
```
Comment (Mingun) [inline bin/peggy-cli.js:218]: Because version 3.0 already released, it's time to remove this option

File: lib/compiler/index.js [hunk 2]
```diff
@@ -125,6 +125,8 @@ const compiler = {
 
     switch (options.output) {
       case "parser":
+        // This is core to this library.
+        // eslint-disable-next-line no-eval
```
Comment (Mingun) [inline lib/compiler/index.js:129]: Just a suggestion. The comment [could be part](https://eslint.org/docs/latest/use/configure/rules#comment-descriptions) of `eslint` command. Probably there exists a tool that can display all such commands with their comments.

File: lib/compiler/passes/generate-bytecode.js [hunk 3]
```diff
@@ -363,6 +363,8 @@ function generateBytecode(ast, options) {
     return buildSequence(
       [op.PUSH_CURR_POS],
       [op.SILENT_FAILS_ON],
+      // Mutual recursion
+      // eslint-disable-next-line no-use-before-define
```
Comment (Mingun) [inline lib/compiler/passes/generate-bytecode.js:367]: Same suggestion here

File: lib/compiler/passes/generate-bytecode.js [hunk 4]
```diff
@@ -519,6 +521,8 @@ function generateBytecode(ast, options) {
     if (delimiterNode) {
       return buildSequence(           //                          stack:[  ]
         [op.PUSH_CURR_POS],           // pos = peg$currPos;       stack:[ pos ]
+        // Mutual recursion
+        // eslint-disable-next-line no-use-before-define
```
Comment (Mingun) [inline lib/compiler/passes/generate-bytecode.js:525]: Same suggestion here

File: lib/compiler/passes/generate-js.js [hunk 5]
```diff
@@ -156,40 +156,40 @@ function generateJS(ast, options) {
...
             + "-"
             + regexpClassEscape(part[1])
-          : regexpClassEscape(part)).join("")
+          : regexpClassEscape(part))).join("")
         + "]/" + (cls.ignoreCase ? "i" : "");
     }
 
-    function buildExpectation(e) {
-      switch (e.type) {
+    function buildExpectation(er) {
```
Comment (Mingun) [inline lib/compiler/passes/generate-js.js:167]: `e` here is a shorthand for `expectation`, so it would be better to rename it to `expectation` / `exp`

**Review by Mingun — APPROVED** [review]
Generally approve, although `no-shadow`, in my mind, are not so useful.

**Comment by hildjj** [comment]
I really like `no-shadow`, but I'll turn it off for now.

File: lib/compiler/passes/generate-bytecode.js [hunk 6]
```diff
@@ -363,6 +363,7 @@ function generateBytecode(ast, options) {
     return buildSequence(
       [op.PUSH_CURR_POS],
       [op.SILENT_FAILS_ON],
+      // eslint-disable-next-line no-use-before-define -- Mutal recursion
```
Comment (polkovnikov-ph) [inline lib/compiler/passes/generate-bytecode.js:366]: \* mutual

File: lib/compiler/passes/generate-js.js [hunk 7]
```diff
@@ -156,12 +156,12 @@ function generateJS(ast, options) {
     function buildRegexp(cls) {
       return "/^["
         + (cls.inverted ? "^" : "")
-        + cls.value.map(part => Array.isArray(part)
+        + cls.value.map(part => (Array.isArray(part)
```
Comment (polkovnikov-ph) [inline lib/compiler/passes/generate-js.js:159]: I'd rather disable the eslint rule instead. IMO this code was completely fine.

File: test/behavior/generated-parser-behavior.spec.js [hunk 8]
```diff
@@ -2894,10 +2895,10 @@ describe("generated parser behavior", () => {
 
         // The "sum" rule
         expect(parser).to.parse("42*43",                   42 * 43);
-        expect(parser).to.parse("42*43+44*45",             42 * 43 + 44 * 45);
-        expect(parser).to.parse("42*43+44*45+46*47+48*49", 42 * 43 + 44 * 45 + 46 * 47 + 48 * 49);
-        expect(parser).to.parse("42*43-44*45",             42 * 43 - 44 * 45);
-        expect(parser).to.parse("42*43-44*45-46*47-48*49", 42 * 43 - 44 * 45 - 46 * 47 - 48 * 49);
+        expect(parser).to.parse("42*43+44*45",             (42 * 43) + (44 * 45));
```
Comment (polkovnikov-ph) [inline test/behavior/generated-parser-behavior.spec.js:2898]: So even basic arithmetic priorities should be highlighted with parens? :)

File: lib/compiler/passes/generate-bytecode.js (hunk 6 above)
Comment (hildjj) [reply lib/compiler/passes/generate-bytecode.js:366]: Fixed all of these, thanks.

File: lib/compiler/passes/generate-js.js (hunk 7 above)
Comment (hildjj) [reply lib/compiler/passes/generate-js.js:159]: I actually like that it's clearer now.

File: test/behavior/generated-parser-behavior.spec.js (hunk 8 above)
Comment (hildjj) [reply test/behavior/generated-parser-behavior.spec.js:2898]: Yep.  I actually agree with this.

**Comment by hildjj** [comment]
note that I messed something up and the tests don't pass.  Fixing that now.

File: bin/peggy-cli.js (hunk 1 above)
Comment (hildjj) [reply bin/peggy-cli.js]: moved to #392

File: lib/compiler/passes/generate-js.js (hunk 7 above)
Comment (hildjj) [reply lib/compiler/passes/generate-js.js:159]: Ironically, I put the parens in the WRONG PLACE the first time. :)

**Review by Mingun — DISMISSED** [review]

**Pull request merged**